### ISSUE TITLE
Using `get` helper with `this.args` results in an error

### ISSUE THREAD

**Issue by miguelcobain** [emberjs/ember.js#18606]
In a glimmer component, when I use
```hbs
{{get this.args "argName"}}
```

results in an error:
```
Assertion Failed: args proxies do not have real property descriptors, so you should never need to call getOwnPropertyDescriptor yourself. This code exists for enumerability, such as in for-in loops and Object.keys()
```

Failing test at #18607

**Comment by rwjblue** [comment]
Awesome, thank you for reporting (and providing the failing test case)!

**Comment by amk221** [comment]
I have a similar problem,

This works ok in the app
```hbs
{{@myComponent.args.foo}}
```

...but doesn't work in the test suite. 

I'm not using the get helper specifically, but I do get the same error

**Comment by locks** [comment]
How are you passing `@myComponent` into that template?

**Comment by amk221** [comment]
A component yields an API to render child components. Those child components render themselves with their parent.

```hbs
{{! parent-component.hbs }}
{{yield (hash Feature=(component "feature" onReady=this.register))}}

  <div class="illustrations">
    {{#each this.features as |feature|}}
      <Illustration @feature={{feature}} />
    {{/each}}
  </div>
```
```hbs
{{! application.hbs }}
<Features as |f|>
  <f.Feature @name="foo" />
  <f.Feature @name="bar" />
</Features>
```
```hbs
{{! (Should) Result in }}
<div class="features">
  <div class="feature">foo</div>
  <div class="feature">bar</div>

  <div class="illustrations"> 
    <div class="illustration illustration--active">foo</div>
    <div class="illustration">bar</div>
  </div>
</div>
```
The use case is an API that allows simple definitions of features, without having to have another API for definition illustrations, which are essentially duplicates.

**Comment by amk221** [comment]
This might be another issue, because the get helper isn't strictly involved. But I get `args proxies do not have real property descriptors` in the test suite, but not in the app. As per above example

**Comment by pzuraq** [comment]
Are you using the latest version of 3.16?

**Comment by amk221** [comment]
Yes

**Comment by amk221** [comment]
I will set up a demo repo

**Comment by amk221** [comment]
I set up an [demo repo], but I only get the error in my application, and not a fresh application.
So, it must be something else. Cheers.

**Comment by amk221** [comment]
My bad,
> Are you using the latest version of 3.16?
I was using `3.16`
but it's fixed in `3.16.3`
